central/auth/service/service_impl_test.go

```diff
@@ -737,6 +738,38 @@ func (s *authServiceAccessControlTestSuite) TestExchangeToken() {
 	s.Equal("sample-token", resp.GetAccessToken())
 }
 
+func (s *authServiceAccessControlTestSuite) TestUpdateRollback() {
+	newConfig := &v1.AuthMachineToMachineConfig{
+		Id:                      "12c053c2-24a7-4b97-bd69-85b3a511241e",
+		TokenExpirationDuration: "1m",
+		Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+	}
+
+	_, err := s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx, &v1.UpdateAuthMachineToMachineConfigRequest{
+		Config: newConfig,
+	})
+	s.Require().NoError(err)
+
+	// No error during rollback.
+	gomock.InOrder(
+		s.mockTokenExchanger.EXPECT().Provider().Return(nil),
+		s.mockIssuerFactory.EXPECT().UnregisterSource(gomock.Any()).Return(nil),
+		s.mockTokenExchanger.EXPECT().Provider().Return(nil),
+		s.mockIssuerFactory.EXPECT().UnregisterSource(gomock.Any()).Return(nil),
+	)
+
+	sameConfig := &v1.AuthMachineToMachineConfig{
+		Id:                      "80c053c2-24a7-4b97-bd69-85b3a511241e",
+		TokenExpirationDuration: "1m",
+		Type:                    v1.AuthMachineToMachineConfig_GITHUB_ACTIONS,
+	}
+	_, err = s.svc.UpdateAuthMachineToMachineConfig(s.accessCtx, &v1.UpdateAuthMachineToMachineConfigRequest{
+		Config: sameConfig,
+	})
+	s.Error(err)
+	s.NotContains(err.Error(), "rollback")
```

Comment:
Can we verify here that rollback worked -> existing token exchanger is in the correct state?